### ISSUE TITLE
Cannot write into "apps" directory

### ISSUE THREAD

**Issue by kerdx** [nextcloud-snap/nextcloud-snap#2162]
After I installed the backup application, it filled up all the server memory after a few days.
It gave me this error: Cannot write into "apps" directory
After clearing the memory on the server by deleting the backups from the terminal I searched online and found this link: https://github.com/nextcloud-snap/nextcloud-snap/issues/1492 my config.php file was also empty. I copied the file at the bottom of the link and changed nextcloud and datadirectory to the default one.
However, nextcloud is not restarting. What can I do?

I'm on ubuntu 18.04

$ snap list nextcloud
Name       Version      Rev    Tracking       Publisher   Notes
nextcloud  24.0.3snap1  31222  latest/stable  nextcloud✓  -

and:

$ snap version
snap    2.56.2
snapd   2.56.2
series  16
ubuntu  18.04
kernel  4.15.0-175-generic

**Comment by PcAndyB** [comment]
exactly the same problem with the snap-install of nextcloud.
Harddisk - no space left because of backup-app backups.
Space free up but no nextcloud: "cannot write apps dir"

**Comment by juliensl** [comment]
Hi everybody ! 

Exactly the same issue for me ! 😢 
Install backup application -> memory full -> clean memory -> Cannot write into "apps" directory -> config.php empty
https://help.nextcloud.com/t/disk-full-after-trying-to-backup/144556 
Nextcloud 24.0.3snap1 too now

I tryied the #1492 too, but it does not work (maybe I did not do the right modifications 🤔 ).

@PcAndyB, could you explain me your Edit1 please ? I don't understand what does it mean, what is former snap ?

Thank you very much.

**Comment by kyrofa** [comment]
Your config is empty? Yikes, that backup app sounds poisonous!

**Comment by kyrofa** [comment]
@kerdx, @juliensl can you please [share logs](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Issue-Debugging-Scripts)?

**Comment by juliensl** [comment]
Hi @kyrofa 

Yes, it is empty : 
```<?php
$CONFIG = array (
  'instanceid' => 'oc8ig6j7n1sq',
);
```

When I diving into `/var/snap/nextcloud/`, I have got the version numbers 29929 and 31222
after `/nextcloud/config/config.php`are both "empty", so if I revert my version lyke @PcAndyB , I think I will have the same problem. 
I can try if you want.

I tried with the config file in #1492 and it does not work, but I am not sure about the modifications to do : 

```
<?php
$snap_name = getenv('SNAP_NAME');

$data_path = '/var/snap/'.$snap_name.'/current';

$database_password = trim(file_get_contents($data_path . '/mysql/nextcloud_password'));

$CONFIG = array (
  'instanceid' => 'oc8ig6j7n1sq',
  'version' => '24.0.3snap1',

  'datadirectory' => '/media/system/data',

  'dbtype' => 'mysql',

  'dbhost' => 'localhost:'.getenv('MYSQL_SOCKET'),

  'dbport' => '',

  'dbname' => 'nextcloud',

  'dbuser' => 'nextcloud',

  'dbpassword' => $database_password,

  'dbtableprefix' => 'oc_',
  'mysql.utf8mb4' => true,

  'installed' => true,

  'apps_paths' => array(
        array(
                'path'=> '/snap/'.$snap_name.'/current/htdocs/apps',
                'url' => '/apps',
                'writable' => false,
        ),

        array(
                'path'=> '/var/snap/'.$snap_name.'/current/nextcloud/extra-apps',
                'url' => '/extra-apps',
                'writable' => true,
        ),
  ),

  'supportedDatabases' => array(
        'mysql',
  ),

  'memcache.locking' => '\OC\Memcache\Redis',
  'memcache.local' => '\OC\Memcache\Redis',
  'redis' => array(
    'host' => getenv('REDIS_SOCKET'),
    'port' => 0,
  ),
);
```

I only changed this : 
```
 'instanceid' => 'oc8ig6j7n1sq',
  'version' => '24.0.3snap1',
```
Do I have to change another thing ?

My Logs attached 
[julienslLogs.txt](https://github.com/nextcloud-snap/nextcloud-snap/files/9511301/julienslLogs.txt)

Note that my server crash in April 2022, But I never had time to repair it ^^'

Thank you very much !

**Comment by kyrofa** [comment]
> so if I revert my version lyke @PcAndyB , I think I will have the same problem.

Indeed, don't do that then. As for the config file you're copying in, check the `'datadirectory' => '/media/system/data',` bit. Is your data actually there? If not, change that to point to where it actually is. If you haven't changed it from the default, then use `/var/snap/nextcloud/common/nextcloud/data`.

**Comment by juliensl** [comment]
Ahhhhh !!! 
Yes, I never change it, and It looks like better now :) 

The apache logs : 
```
root@ns379063:/var/snap/nextcloud/common/nextcloud# sudo snap logs -f nextcloud.apache
2022-09-07T22:59:23+02:00 nextcloud.apache[29996]: System config value memcache.locking set to string \OC\Memcache\Redis
2022-09-07T22:59:24+02:00 nextcloud.apache[29996]: Nextcloud or one of the apps require upgrade - only a limited number of commands are available
2022-09-07T22:59:24+02:00 nextcloud.apache[29996]: You may use your browser or the occ upgrade command to do the upgrade
2022-09-07T22:59:24+02:00 nextcloud.apache[29996]: System config value memcache.local set to string \OC\Memcache\Redis
2022-09-07T22:59:25+02:00 nextcloud.apache[29996]: Nextcloud or one of the apps require upgrade - only a limited number of commands are available
2022-09-07T22:59:25+02:00 nextcloud.apache[29996]: You may use your browser or the occ upgrade command to do the upgrade
2022-09-07T22:59:25+02:00 nextcloud.apache[29996]: No such app enabled: updatenotification
2022-09-07T22:59:26+02:00 nextcloud.apache[29996]: Nextcloud or one of the apps require upgrade - only a limited number of commands are available
2022-09-07T22:59:26+02:00 nextcloud.apache[29996]: You may use your browser or the occ upgrade command to do the upgrade
2022-09-07T22:59:26+02:00 nextcloud.apache[29996]: Making sure nextcloud is fully upgraded...
2022-09-07T22:59:27+02:00 nextcloud.apache[29996]: Nextcloud or one of the apps require upgrade - only a limited number of commands are available
2022-09-07T22:59:27+02:00 nextcloud.apache[29996]: You may use your browser or the occ upgrade command to do the upgrade
2022-09-07T22:59:27+02:00 nextcloud.apache[29996]: Setting log level to debug
2022-09-07T22:59:27+02:00 nextcloud.apache[29996]: Turned on maintenance mode
2022-09-07T22:59:34+02:00 nextcloud.apache[29996]: Updating database schema
2022-09-07T23:00:32+02:00 nextcloud.apache[29996]: Updated database
2022-09-07T23:00:32+02:00 nextcloud.apache[29996]: Disabled incompatible app: backup
2022-09-07T23:00:32+02:00 nextcloud.apache[29996]: Disabled incompatible app: bruteforcesettings
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Disabled incompatible app: metadata
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updating <federation> ...
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updated <federation> to 1.14.0
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updating <lookup_server_connector> ...
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updated <lookup_server_connector> to 1.12.0
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updating <oauth2> ...
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updated <oauth2> to 1.12.0
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updating <password_policy> ...
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updated <password_policy> to 1.14.0
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updating <files> ...
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updated <files> to 1.19.0
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updating <activity> ...
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updated <activity> to 2.16.0
2022-09-07T23:00:33+02:00 nextcloud.apache[29996]: Updating <circles> ...
2022-09-07T23:00:59+02:00 nextcloud.apache[29996]: Updated <circles> to 24.0.0
2022-09-07T23:00:59+02:00 nextcloud.apache[29996]: Updating <cloud_federation_api> ...
2022-09-07T23:00:59+02:00 nextcloud.apache[29996]: Updated <cloud_federation_api> to 1.7.0
2022-09-07T23:00:59+02:00 nextcloud.apache[29996]: Updating <dav> ...
2022-09-07T23:01:00+02:00 nextcloud.apache[29996]: Fix broken values of calendar objects
2022-09-07T23:01:00+02:00 nextcloud.apache[29996]:  Starting ...
2022-09-07T23:01:00+02:00 nextcloud.apache[29996]:     0/0 [>---------------------------]   0%
2022-09-07T23:01:00+02:00 nextcloud.apache[29996]: Clean up old calendar subscriptions from deleted users that were not cleaned-up
2022-09-07T23:01:00+02:00 nextcloud.apache[29996]:  Starting ...
2022-09-07T23:01:00+02:00 nextcloud.apache[29996]:     0/0 [->--------------------------]   0%
2022-09-07T23:01:00+02:00 nextcloud.apache[29996]: Updated <dav> to 1.22.0
2022-09-07T23:01:00+02:00 nextcloud.apache[29996]: Updating <files_external> ...
2022-09-07T23:01:08+02:00 nextcloud.apache[29996]: Updated <files_external> to 1.16.1
2022-09-07T23:01:08+02:00 nextcloud.apache[29996]: Updating <files_sharing> ...
2022-09-07T23:01:23+02:00 nextcloud.apache[29996]: Updated <files_sharing> to 1.16.2
2022-09-07T23:01:23+02:00 nextcloud.apache[29996]: Updating <files_trashbin> ...
2022-09-07T23:01:23+02:00 nextcloud.apache[29996]: Updated <files_trashbin> to 1.14.0
2022-09-07T23:01:23+02:00 nextcloud.apache[29996]: Updating <files_versions> ...
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updated <files_versions> to 1.17.0
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updating <sharebymail> ...
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updated <sharebymail> to 1.14.0
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updating <workflowengine> ...
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updated <workflowengine> to 2.6.0
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updating <comments> ...
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updated <comments> to 1.14.0
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updating <firstrunwizard> ...
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updated <firstrunwizard> to 2.13.0
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updating <nextcloud_announcements> ...
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updated <nextcloud_announcements> to 1.13.0
2022-09-07T23:01:24+02:00 nextcloud.apache[29996]: Updating <notifications> ...
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]: Updated <notifications> to 2.12.0
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]: Updating <systemtags> ...
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]: Updated <systemtags> to 1.14.0
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]: Updating <accessibility> ...
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]: Migrate old user config
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]:  Starting ...
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]:     0/0 [--->------------------------]   0%
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]: Updated <accessibility> to 1.10.0
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]: Updating <contactsinteraction> ...
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]: Updated <contactsinteraction> to 1.5.0
2022-09-07T23:01:25+02:00 nextcloud.apache[29996]: Updating <dashboard> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <dashboard> to 7.4.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <federatedfilesharing> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <federatedfilesharing> to 1.14.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <files_pdfviewer> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <files_pdfviewer> to 2.5.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <files_rightclick> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <files_rightclick> to 1.3.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <files_videoplayer> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <files_videoplayer> to 1.13.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <logreader> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <logreader> to 2.9.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <photos> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <photos> to 1.6.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <privacy> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <privacy> to 1.8.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <provisioning_api> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <provisioning_api> to 1.14.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <recommendations> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <recommendations> to 1.3.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <serverinfo> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <serverinfo> to 1.14.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <settings> ...
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updated <settings> to 1.6.0
2022-09-07T23:01:26+02:00 nextcloud.apache[29996]: Updating <support> ...
2022-09-07T23:01:27+02:00 nextcloud.apache[29996]: Updated <support> to 1.7.0
2022-09-07T23:01:27+02:00 nextcloud.apache[29996]: Updating <survey_client> ...
2022-09-07T23:01:27+02:00 nextcloud.apache[29996]: Updated <survey_client> to 1.12.0
2022-09-07T23:01:27+02:00 nextcloud.apache[29996]: Updating <text> ...
2022-09-07T23:01:33+02:00 nextcloud.apache[29996]: Updated <text> to 3.5.1
2022-09-07T23:01:33+02:00 nextcloud.apache[29996]: Updating <twofactor_backupcodes> ...
2022-09-07T23:01:33+02:00 nextcloud.apache[29996]: Updated <twofactor_backupcodes> to 1.13.0
2022-09-07T23:01:33+02:00 nextcloud.apache[29996]: Updating <user_status> ...
2022-09-07T23:01:33+02:00 nextcloud.apache[29996]: Updated <user_status> to 1.4.0
2022-09-07T23:01:33+02:00 nextcloud.apache[29996]: Updating <viewer> ...
2022-09-07T23:01:33+02:00 nextcloud.apache[29996]: Updated <viewer> to 1.8.0
2022-09-07T23:01:33+02:00 nextcloud.apache[29996]: Updating <weather_status> ...
2022-09-07T23:01:33+02:00 nextcloud.apache[29996]: Updated <weather_status> to 1.4.0
2022-09-07T23:01:42+02:00 nextcloud.apache[29996]: Update app backup from App Store
2022-09-07T23:01:47+02:00 nextcloud.apache[29996]: Update app bruteforcesettings from App Store
2022-09-07T23:01:48+02:00 nextcloud.apache[29996]: Update app metadata from App Store
2022-09-07T23:01:58+02:00 nextcloud.apache[29996]: Starting code integrity check...
2022-09-07T23:02:43+02:00 nextcloud.apache[29996]: Finished code integrity check
2022-09-07T23:02:43+02:00 nextcloud.apache[29996]: Update successful
2022-09-07T23:02:43+02:00 nextcloud.apache[29996]: Turned off maintenance mode
2022-09-07T23:02:43+02:00 nextcloud.apache[29996]: Resetting log level
2022-09-07T23:02:43+02:00 nextcloud.apache[29996]: <warning>The setting "trusted_domains" could not be set automatically by the upgrade script, please set it manually</warning>All set! Running httpd...
2022-09-07T23:02:43+02:00 nextcloud.apache[29996]: No certificates are active: using HTTP only
2022-09-07T23:02:44+02:00 nextcloud.apache[29996]: HTTP compression is disabled
```

And I've got this message on chrome : 
```
Accès à partir d'un domaine non approuvé
Veuillez contacter votre administrateur. Si vous êtes un administrateur, éditez la variable "trusted_domains" dans le fichier config/config.php comme l'exemple dans le fichier config/config.sample.php.


Vous trouverez d'autres informations sur la configuration dans la [documentation ](https://docs.nextcloud.com/server/24/go.php?to=admin-trusted-domains).
```

I guess I just have to setup the SSL again, I will check the documentation, thank you very much !!!

Last question, I used this backup app to migrate my server from a self-hosted server to a fully-managed one.
Actually, I only used nextcloud to store files and we are only 2 users.
If I download this "data" file and upload it to the new service with ftp, my transfert will be done, exactly ? 
Folder architecture looks like ok, do I forget something else ? 

Thank you again :)

**Comment by kyrofa** [comment]
If:

1. You're only using the files app (i.e. no calendar, no contacts, nothing else)
2. You grab `/var/snap/nextcloud/common/nextcloud/data`

Then yes, you have everything you need. That will include all the files for all the users. It just won't include any database things, like access times, sharing links, calendar, contacts, and so on.

**Comment by kyrofa** [comment]
I'm going to go ahead and close this, as I believe this is caused by the backup app and hopefully how to (mostly) recover is clear, now. Can someone please log an issue against that app you're using, please? Something seems badly broken there.

**Comment by juliensl** [comment]
My debug is not completly finished, I had to : 

- Setup my trusted domain : `sudo nextcloud.occ config:system:set trusted_domains 1 --value=example.com`
- Exit of maintenance mode : `sudo nextcloud.occ maintenance:mode --off`
- Setup my SSL again : `sudo ufw allow 80,443/tcp` then `sudo nextcloud.enable-https lets-encrypt`

And after, when I try to log me in, I have : 
```
Erreur interne du serveur
Le serveur est incapable d'exécuter votre requête.

Si cela se reproduit, veuillez envoyer les détails techniques ci-dessous à l'administrateur du serveur.

Le fichier journal du serveur peut fournir plus de renseignements.

Renseignements techniques
Adresse distante : my-ip
ID de la demande : 65GrprvZQ8MMvhnoopLh
```

This is an "internal server error, server cannot execute the request".
I tried to reset my password with `sudo nextcloud.occ user:resetpassword my-username` but it does not solve the problem.
Actually, not need to try to continue to debug it, I will make the transfert of the data folder with FTP and I will kill this server after.
But for other people, maybe they will have the same problem ;)

Thx again @kyrofa :) 

I will post an issue on the github repo ;) 

++
